tests/digraph/test_all_simple_paths.py

```diff
@@ -111,3 +111,214 @@ def test_graph_digraph_all_simple_paths(self):
         dag.add_node(0)
         dag.add_node(1)
         self.assertRaises(TypeError, retworkx.digraph_all_simple_paths, (dag, 0, 1))
+
+
+class TestDAGAllSimplePathsAllPairs(unittest.TestCase):
```

Comment:
```suggestion
class TestDiGraphAllSimplePathsAllPairs(unittest.TestCase):
```
since it seems that there is a cycle `[(1, 2), (2, 3), (3, 4), (4, 2)]` in the testing graph and hence not a DAG.